assignment3/WordFinder.java

```diff
@@ -0,0 +1,91 @@
+package com.katermar.assignment3;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Set;
+
+/**
+ * Created by katermar on 6/8/2017.
+ */
+
+/*
+class WordFinder to find words in a char grid
+ */
+public class WordFinder {
+    private Dictionary dictionary;

```

Comment:
Could be final, as `setDictionary` is not really needed. It is better to initialize classes once in constructor and keep them immutable afterwards. It reduces code complexity and lets you avoid serious bugs, especially if the class will ever be used in multi-threaded environment.

If you need to find words from a different dictionary in the future, it is cheap to just create new `WordFinder` instance initialized with new Dictionary.